src/Runner.Common/HostContext.cs

```diff
@@ -221,8 +221,11 @@ public HostContext(string hostType, string logFile = null)
             if (File.Exists(runnerFile))
             {
                 var runnerSettings = IOUtil.LoadObject<RunnerSettings>(runnerFile);
-                _userAgents.Add(new ProductInfoHeaderValue("RunnerId", runnerSettings.AgentId.ToString(CultureInfo.InvariantCulture)));
-                _userAgents.Add(new ProductInfoHeaderValue("GroupId", runnerSettings.PoolId.ToString(CultureInfo.InvariantCulture)));
+                if (runnerSettings != null)
```

Comment:
should we just throw and crash the runner if `runnerSettings` is NULL?

Reply:
Thanks!